ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/CastorFullOptimization.java

```diff
@@ -379,7 +379,7 @@ static boolean shouldRunSecondPreventiveRao(RaoParameters raoParameters, Objecti
      * Returns true if any result has a positive functional cost
      */
     private static boolean isAnyResultUnsecure(Collection<OptimizationResult> results) {
-        return results.stream().anyMatch(optimizationResult -> optimizationResult.getFunctionalCost() >= 0);
+        return results.stream().anyMatch(optimizationResult -> optimizationResult.getFunctionalCost() >= 0 || optimizationResult.getVirtualCost() > 1e-6);
```

Comment:
It would be more consistent to put "> 1e-6" on both conditions (functional cost and virtual cost)

Reply:
I wanted to match the SearchTree stop criterion (0 is not considered secure)

Reply:
Okay. I forgot virtual costs cannot go below 0 whereas functional costs can, so it indeed makes sense to have different conditions